course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -74,61 +56,632 @@ describe('Unit testing PeopleController', function() {
+            expect(result).toBeNull();
+        });
+        it('won\'t add if they could not be found via /people lookup', function(){
+            var personResult = [{}, ["test45@mcelroy.org"]];
+            spyOn(scope, 'filterSearchResults').and.returnValue([]);
+            spyOn(scope, 'getProfileFullName').and.returnValue('Test User');
+            spyOn(scope, 'addNewMemberToCourse').and.returnValue({'result': 'success'});
+            var result = scope.addNewMember(personResult, members);
+            expect(scope.messages.warnings).toEqual([{type: 'notFound',
+                    searchTerm: personResult[1]}]);
+            expect(scope.getProfileFullName).not.toHaveBeenCalled();
+
+            expect(result).toBeNull();
+        });
+        it('won\'t add if multiple profiles returned by /people lookup', function(){
+            var secondPerson = {
+                    active: 1,
+                    email_address: "test48@mcelroy.org",
+                    name_first: "Test",
+                    name_last: "User2",
+                    prime_role_indicator: "Y",
+                    role_type_cd: "STUDENT",
+                    univ_id: "12345675"
+                };
+            spyOn(scope, 'filterSearchResults').and.returnValue([personResult[0], secondPerson]);
+            spyOn(scope, 'getProfileFullName').and.returnValue('Test User');
+            spyOn(scope, 'addNewMemberToCourse').and.returnValue({'result': 'success'});
+            var result = scope.addNewMember(personResult, members);
+            expect(scope.messages.warnings[0].type).toEqual('multipleProfiles');
+            expect(scope.getProfileFullName).toHaveBeenCalledWith(personResult[0]);
+            expect(result).toBeNull();
+        });
+        it('adds person if not in course and single profile available', function(){
+            var newPerson = [
+                {
+                    active: 1,
+                    email_address: "test49@mcelroy.org",
+                    name_first: "Test",
+                    name_last: "New User",
+                    prime_role_indicator: "Y",
+                    role_type_cd: "STUDENT",
+                    univ_id: '12345674'
+                },
+                ["test49@mcelroy.org"]
+            ];
+            spyOn(scope, 'filterSearchResults').and.returnValue([newPerson[0]]);
+            spyOn(scope, 'getProfileFullName').and.returnValue('Test New User');
+            spyOn(scope, 'addNewMemberToCourse').and.returnValue({'result': 'success'});
+            var result = scope.addNewMember(newPerson, members);
+            var postParams = {
+                        user_id: '12345674',
+                        role_id: 0};
+            expect(scope.getProfileFullName).toHaveBeenCalledWith(newPerson[0]);
+            expect(scope.addNewMemberToCourse).toHaveBeenCalledWith(postParams, 'Test New User', newPerson[1]);
+            expect(result).toEqual({ result: 'success' });
```

Comment:
I guess it doesn't matter, since we're returning whatever `addNewMemberToCourse()` provides.